lighthouse-cli/performance-experiment/formatters/partials/config-panel.html

```diff
@@ -0,0 +1,286 @@
+    {{else}}
+      <span class="tree-marker vert-right"></span>
+      <span class="tree-marker right"></span>
+    {{/isLastChild}}
+    </span>
+
+    <button class="request-blocking-toggle js-request-blocking-toggle"></button>
+    <span class="cnc-node__tree-value">
+      {{#parseURL this.node.request.url }}
+        <span class="cnc-node__tree-file">{{ this.file }}</span>
+        <span class="cnc-node__tree-hostname">({{ this.hostname }})</span>
+      {{/parseURL}}
+      {{#unless hasChildren}}
+        - <span class="cnc-node__chain-duration">{{chainDuration startTime this.node.request.endTime }}ms, {{formatTransferSize this.transferSize}}KB</span>
+      {{/unless}}
+    </span>
+
+    {{#each this.node.children as |child| }}
+      {{#createContextFor ../node.children @key ../treeMarkers ../isLastChild ../startTime ../transferSize }}
+        {{> writeNode this }}
+      {{/createContextFor }}
+    {{/each}}
+  </div>
+{{/inline}}
+<div class="config-panel js-config-panel">
+  <div class="config-panel__body">
+    <div class="config-panel__tabs js-tabs">
+      <div class="config-panel__tabs__tab-bar">
+          <a href="#config-panel__tabs__list-panel" class="config-panel__tabs__tab js-tab is-active">List View</a>
+          <a href="#config-panel__tabs__tree-panel" class="config-panel__tabs__tab js-tab">Tree View</a>
+      </div>
+
+      <div class="config-panel__tabs__tab-panel js-tab-panel is-active" id="config-panel__tabs__list-panel">
+        <ul class="url-blocking-patterns js-url-blocking-patterns">
+          <template class="url-blocking-entry">
+            <li class="url-blocking-entry" data-url-pattern="">
+              <button class="url-blocking-entry__cancel-button"></button>
+              <div class="url-blocking-entry__pattern"></div>
+            </li>
+          </template>
+          <li class="url-blocking-entry">
+            <button class="url-blocking-entry__add-button js-add-button"></button>
+            <input class="url-blocking-entry__pattern js-pattern-input" placeholder="URL blocking Patterns (* for wild card) e.g. *://www.evil.com/*">
+          </li>
+        </ul>
+      </div>
+
+      <div class="config-panel__tabs__tab-panel js-tab-panel" id="config-panel__tabs__tree-panel">
+        <div>Initial navigation</div>
+        {{#createTreeRenderContext this}}
+          {{#each this.tree }}
+            {{#createContextFor ../tree @key undefined undefined ../startTime ../transferSize }}
+              {{> writeNode this }}
+            {{/createContextFor}}
+          {{/each}}
+        {{/createTreeRenderContext}}
+      </div>
+
+    </div>
+  </div>
+  <div class="config-panel__header">
+    <button class="config-panel__panel-toggle js-panel-toggle"></button>
+    <div class="config-panel__message js-message"></div>
+    <button class="js-rerun-button">Rerun</button>
+  </div>
+</div>
+
+<script>
+window.addEventListener('DOMContentLoaded', () => {
```

Comment:
I'm not sure we want to put script in a partial formatter. As of now, our partial have just been dumb render targets for data. This one is different in that it adds functionality. The script will also be injected every time the partial is included in the report. Presumably that's one time, but it would be good not to go that route :)

Is it possible to do this in your main perf-x.js script?